Specs/Core/ScreenSpaceEventHandlerSpec.js

```diff
@@ -1037,6 +1037,113 @@ defineSuite([
         expect(action).not.toHaveBeenCalled();
     });
 
+    it('handles touch pinch release', function() {
+        var leftDownEventType = ScreenSpaceEventType.LEFT_DOWN;
+        var leftDownAction = createCloningSpy('LEFT_DOWN');
+        handler.setInputAction(leftDownAction, leftDownEventType);
+
+        var pinchStartEventType = ScreenSpaceEventType.PINCH_START;
+        var pinchStartAction = createCloningSpy('PINCH_START');
+        handler.setInputAction(pinchStartAction, pinchStartEventType);
+
+        var pinchEndEventType = ScreenSpaceEventType.PINCH_END;
+        var pinchEndAction = createCloningSpy('PINCH_END');
+        handler.setInputAction(pinchEndAction, pinchEndEventType);
+
+        var touch1Position = {
+            clientX : 1,
+            clientY : 2
+        };
+        var touch2Position = {
+            clientX : 4,
+            clientY : 3
+        };
+        if (usePointerEvents) {
+            DomEventSimulator.firePointerDown(element, combine(
+                {
+                    pointerType : 'touch',
+                    pointerId : 1
+                },
+                touch1Position
+            ));
+            DomEventSimulator.firePointerDown(element, combine(
+                {
+                    pointerType : 'touch',
+                    pointerId : 2
+                },
+                touch2Position
+            ));
+
+            // Releasing one of two fingers should not trigger
+            // PINCH_END or LEFT_DOWN:
+            leftDownAction.calls.reset();
+            DomEventSimulator.firePointerUp(element, combine(
+                {
+                    pointerType : 'touch',
+                    pointerId : 1
+                },
+                touch1Position
+            ));
+            expect(pinchEndAction).not.toHaveBeenCalled();
+            expect(leftDownAction).not.toHaveBeenCalled();
+            pinchEndAction.calls.reset();
+
+            // Putting another finger down should not trigger
+            // PINCH_START:
```

Comment:
This test fails, and indeed `PINCH_START` is re-called when the user's touch count deviates from 2 (while remaining non-zero) and returns to 2, bringing pinch out of suspension.  I'm not sure this is a bad thing, as the pinch is re-starting even though it never fully stopped.

Anyway the test should be fixed, not sure if by changing the test's expectation or by suppressing the pinch re-start event.

Reply:
Thanks – I will look into this next week!

Reply:
Since we aren't firing PINCH_END with a single touch up, it indeed doesn't make sense to fire PINCH_START when moving back to 2 fingers down.

I added a check to see if we are already pinching when touch count returns to 0.. All tests should be green and I merged in master.